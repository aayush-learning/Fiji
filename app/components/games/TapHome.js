import React, { Component } from 'react';
import { View, Text } from 'react-native';
import Confirm from './Confirm';

export default class TapHome extends Component {
  
  constructor(){
    super();
    this.state={
      // This is our Default number value
      numberHolder : 4, 
      counter : 0,
      count : '',
      showModal: false
    }
  }

  componentDidMount() {

    //This will start timer and will update text value
    setInterval(() => {
      let temp = this.state.count - this.state.numberHolder
      if((this.state.count - this.state.numberHolder) > 2){
        this.setState({count:  this.state.numberHolder -  3 })
      }

      if((this.state.numberHolder - this.state.count) > 2){
        this.setState({count: this.state.numberHolder -  2 })
      }

      else{
        this.setState({count: this.state.count + 1})
      }
    }, 2000)
  }
 
  //This will generate random number and will check on tap condition
  GenerateRandomNumber=()=>
  {
    if( this.state.numberHolder == this.state.count ) {
      var RandomNumber = Math.floor(Math.random() * 100) + 1 ;
      if( RandomNumber > 9 ){
        RandomNumber = RandomNumber % 10;
      }

      if( RandomNumber == 0 ){
        RandomNumber = RandomNumber + 1;
      }
      if( RandomNumber == 1 ){
        this.setState({
          count: 0
        })
      }
      if( RandomNumber == 2 ){
        this.setState({
          count: 0
        })
      }
      this.setState({
      numberHolder : RandomNumber,
      })
    }
    else {
      this.setState({
       showModal: !this.state.showModal 
      })
    }
  }

  onAccept() {

  }

  onDecline() {
   this.setState({ showModal: false });
   this.props.navigate('Game2');
  }

 render() {
    const { container, circle, text, subText} = styles;
    const {count, numberHolder} = this.state

   // const {goBack} = this.props.navigation;

    return (
      <View style={container} >
        <View style={circle}>
          <View>
            <Text style={text}>{numberHolder}</Text>
          </View>
        </View> 
        <Text style={subText} onPress={this.GenerateRandomNumber}>{count}</Text>
        <Confirm
          visible={this.state.showModal}
          onAccept={this.onAccept.bind(this)}
          onDecline={this.onDecline.bind(this)}
        >
          OPS! TOO LATE
        </Confirm>
      </View>
    );
  }
}//End of class component

const styles = {
  container: {
    flex: 1,
    backgroundColor: '#576A6E'
  },
  circle: {
    width: 120,
    height: 120,
    borderRadius: 120/2,
    borderWidth: 5,
    borderColor: 'white',
    justifyContent: 'center',
    alignSelf: 'center',
    marginTop: '20%'
  },
  text: {
    fontFamily: 'Cochin',
    fontSize: 60,
    color: 'white',
    justifyContent: 'center',
    alignSelf: 'center',
  },
  subText: {
    fontFamily: 'Cochin',
    fontSize: 90,
    color: 'white',
    justifyContent: 'center',
    alignSelf: 'center',
    marginTop:'10%'
  },
};//End of styles