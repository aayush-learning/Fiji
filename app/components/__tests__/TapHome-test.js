import 'react-native';
import React from 'react';
import { Dimensions } from 'react-native'
import TapHome from '../games/TapHome'
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const window = Dimensions.get("window")

it('renders correctly', () => {
  const tree = renderer.create(
    <TapHome 
        data={{
            answer: 15,
            serial: ['9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20'],
        }}
        style = {{
            width: window.width,
            height: window.height
        }}
        setProgress={()=>{}}
        onScore={()=>{}}
        onEnd={()=>{}}  
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
