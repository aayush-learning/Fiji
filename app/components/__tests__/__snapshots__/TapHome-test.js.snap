// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "paddingTop": 266.8,
      },
    ]
  }
>
  <View
    collapsable={undefined}
    style={Object {}}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "height": 194,
          "width": 194,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#24B2EA",
            "borderRadius": 8,
            "elevation": 8,
            "height": 194,
            "justifyContent": "center",
            "position": "absolute",
            "shadowColor": "grey",
            "shadowOpacity": 1,
            "shadowRadius": 8,
            "width": 194,
          }
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#FFFFFF",
              "fontSize": 154,
            }
          }
        >
          15
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontFamily": "Cochin",
            "justifyContent": "center",
          },
          Object {
            "fontSize": 169,
            "marginTop": 133.4,
          },
        ]
      }
    >
      9
    </Text>
  </View>
</View>
`;
